### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by RomanDorosinec**

### CODE REVIEW

File: taskfour/src/by/dorosinets/training/quadraticequation/Runner.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Empty parameters");
+            // Exception processing if entered string
+        } catch (NumberFormatException e) {
+            System.out.println("Not valid format number");
+        }
+    }
+
+    /**
+     * Check our arguments on the amount of numbers after the decimal point
```
Comment (AlexanderSoroka) [inline taskfour/src/by/dorosinets/training/quadraticequation/Runner.java]: What kind of issue are you trying to solve with this check?

Comment (AlexanderSoroka) [reply taskfour/src/by/dorosinets/training/quadraticequation/Runner.java]: and it's actually not valid - what if user input 10e-15 - it's around 15 numbers after decimal point but your check will definitely miss it

File: taskfour/src/by/dorosinets/training/quadraticequation/Equation.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public final static int MAX_SIZE = 4;
+
+    /**
+     * Creates new equation
+     * @param a coefficient near x^2
+     * @param b coefficient near x
+     * @param c free coefficient
+     */
+    public Equation(double a, double b, double c) {
+        this.numberA = a;
```
Comment (AlexanderSoroka) [inline taskfour/src/by/dorosinets/training/quadraticequation/Equation.java]: what if somebody pass to this class a == NaN?

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: taskfour/src/by/dorosinets/training/quadraticequation/Equation.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Creates new equation
+     * Tests for values NaN
+     * @param a coefficient near x^2
+     * @param b coefficient near x
+     * @param c free coefficient
+     */
+    public Equation(double a, double b, double c) {
+        if (Double.isNaN(a) || Double.isNaN(b) || Double.isNaN(c)) {
```
Comment (AlexanderSoroka) [inline taskfour/src/by/dorosinets/training/quadraticequation/Equation.java]: So infinities are accepted values?

File: taskfour/src/by/dorosinets/training/quadraticequation/Equation.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        this.numberC = c;
+    }
+
+    /**
+     * Calculates equation roots
+     * @return a string with equation solution - no roots, one root or two roots
+     */
+    public String rootEquation() {
+        // Handles the case when coefficient A is zero
+        if (numberA == 0.0) {
```
Comment (AlexanderSoroka) [inline taskfour/src/by/dorosinets/training/quadraticequation/Equation.java]: Invalid double comparison

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
I don't see task#5 in this pull request? Where can I take a look on it?

File: taskfour/src/by/dorosinets/training/quadraticequation/Equation.java (hunk 4 above)
Comment (RomanDorosinec) [inline taskfour/src/by/dorosinets/training/quadraticequation/Equation.java]: I know, but I don't have free time, chenge this mistake. I'm so sorry

**Pull request merged**